vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -0,0 +1,418 @@
+            Pandas object is a DataFrame.
+
+        Returns
+        -------
+
+        pd.DataFrame
+            The returned dataframe will contain labels. If more than one image in input,
+            the returned dataframe will contain `images` column, and all results are concatenated.
+        """
+        if self._classifier is None:
+            raise RuntimeError('Classifier is not initialized, try `fit` first.')
+        proba = self._classifier.predict(data)
+        if 'image' in proba.columns:
+            # multiple images
+            ret = proba.loc[proba.groupby(["image"])["score"].idxmax()].reset_index(drop=True)
+        else:
+            # single image
+            ret = proba.loc[[proba["score"].idxmax()]]
+        if as_pandas:
+            return ret
+        else:
+            return ret.as_numpy()
+
+    def predict_feature(self, data, as_pandas=True):
+        """Predict images visual feature representations, return the features as numpy (1xD) vector.
+
+        Parameters
+        ----------
+        data : str, pd.DataFrame or ndarray
+            The input, can be str(filepath), pd.DataFrame with 'image' column, or raw ndarray input.
+        as_pandas : bool, default = True
+            Whether to return the output as a pandas object (True) or list of numpy array(s) (False).
+            Pandas object is a DataFrame.
+
+        Returns
+        -------
+
+        pd.DataFrame
+            The returned dataframe will contain image features. If more than one image in input,
+            the returned dataframe will contain `images` column, and all results are concatenated.
+        """
+        if self._classifier is None:
+            raise RuntimeError('Classifier is not initialized, try `fit` first.')
+        ret = self._classifier.predict_feature(data)
+        if as_pandas:
+            return ret
+        else:
+            return ret.as_numpy()
+
+    def evaluate(self, data):
+        """Evaluate model performance on validation data.
+
+        Parameters
+        ----------
+        data : pd.DataFrame or iterator
+            The validation data.
+        """
+        if self._classifier is None:
+            raise RuntimeError('Classifier not initialized, try `fit` first.')
+        return self._classifier.evaluate(data)
+
+    def fit_summary(self):
+        """Return summary of last `fit` process.
+
+        Returns
+        -------
+        dict
+            The summary of last `fit` process. Major keys are ('train_acc', 'val_acc', 'total_time',...)
+
+        """
+        return copy.copy(self._fit_summary)
+
+    def save(self, file_name):
+        """Dump predictor to disk.
+
+        Parameters
+        ----------
+        file_name : str
+            The file name of saved copy.
+
+        """
+        with open(file_name, 'wb') as fid:
+            pickle.dump(self, fid)
+
+    @classmethod
+    def load(cls, file_name, verbosity=2):
```

Comment:
`file_name` -> `path`, ideally it should be the same path value as specified by user or available via `predictor.path`.